game-core/src/main/java/games/strategy/triplea/ui/FlagDrawMode.java

```diff
@@ -0,0 +1,63 @@
+package games.strategy.triplea.ui;
+
+import games.strategy.triplea.ui.screen.UnitsDrawer;
+import java.util.prefs.Preferences;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+/**
+ * Class to change how unit flags are drawn on the map. Unit flags are flags drawn underneath or
+ * below units to make it easier to identify which nation they belong to. By default flag drawing is
+ * turned off, toggling once turns small flags on, toggle again to turn on large flags and again to
+ * turn off.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+class FlagDrawMode {
+
+  static void toggleNextDrawMode(final MapPanel mapPanel) {
+
+    final Preferences prefs = Preferences.userNodeForPackage(FlagDrawMode.class);
+    final UnitsDrawer.UnitFlagDrawMode setting =
+        Enum.valueOf(
+            UnitsDrawer.UnitFlagDrawMode.class,
+            prefs.get(
+                UnitsDrawer.PreferenceKeys.DRAW_MODE.name(),
+                UnitsDrawer.UnitFlagDrawMode.SMALL_FLAG.toString()));
+    UnitsDrawer.setUnitFlagDrawMode(setting, prefs);
+    UnitsDrawer.enabledFlags =
+        prefs.getBoolean(
+            UnitsDrawer.PreferenceKeys.DRAWING_ENABLED.name(), UnitsDrawer.enabledFlags);
+
+    // draw mode is off, toggle small flag
+    if (!prefs.getBoolean(UnitsDrawer.PreferenceKeys.DRAWING_ENABLED.name(), false)) {
+      UnitsDrawer.enabledFlags = true;
+      prefs.putBoolean(UnitsDrawer.PreferenceKeys.DRAWING_ENABLED.name(), true);
+      prefs.put(
+          UnitsDrawer.PreferenceKeys.DRAW_MODE.name(),
+          UnitsDrawer.UnitFlagDrawMode.SMALL_FLAG.toString());
```

Comment:
Same here with name VS toString

Reply:
Only need to point out items once. I'm looking for a good reading that went into detail and am turning up a blank. I recall the reasoning was to avoid the 'avalance' effect of many comments. Leaving a review that has many, many comments works against the reviews goals and is subtly saying "you did a shit job, you don't know how to code". I wish I could find that resource for the reading, it was interesting.

This section had some interesting and pertinent points IMO as well:
> https://blog.asana.com/2016/12/7-ways-to-uplevel-your-code-review-skills/
> 7. Always give approval, unless you can prove that there is a bug

The UX change here is significant though, it's okay to let a review be open for a bit to fully consider it and to simply state that as well.

Reply:
True, but in this case this is my own code I wrote almost 3 years in the past 😅 
I agree that not all of those comments are necessary, but I felt like there were some hidden toString statements in unexpected places

Reply:
Understood, the thing to be careful about is the perception of "Oh, I know this code, it's mine and it is not great. Could you please do these extra things beyond the scope of your goal to make improvements I should have made some time ago? You know, while you're here.."

To be clear, I think I understand your motivation and I appreciate improving the code. Just want to convey that it runs a risk of the above perception. 

At the same time it's difficult to know how exactly we clean up the TripleA codebase. I think the answer is to move faster and pick our battles. We perhaps should let smaller things slide and do a better job of adding tests and larger rewrites where most of these points could be mooted. The danger I see is moving slowly on small cleanups that would be removed with a more significant rewrite. At the same time, it's hard to know if we'll ever get those tests and the more significant rewrite. It is a balance I suppose that is not easy to strike.